internal/filter/filter.go

```diff
@@ -7,32 +7,38 @@ import (
 
 var matchGroups = regexp.MustCompile(`(?:([^\s"]+)|"([^"]*)"?)+`)
 
-// Split terms by spaces, except for spaces within quotes.
-// Like http://stackoverflow.com/questions/16261635/javascript-split-string-by-space-but-ignore-space-in-quotes-notice-not-to-spli
-// except:
-// - Treat the query ["san fran] as having the single term ["san fran"] instead of ["san", "fran"]
-// - Removes the delimiting quotes.
-func Match(str, query string) bool {
+func GetMatchStrings(query string) []string {
 	if query == "" {
-		return true
+		return nil
 	}
-
 	matches := matchGroups.FindAllStringSubmatch(query, -1)
-	for _, match := range matches {
+	matchStrings := make([]string, len(matches), len(matches))
+	for i, match := range matches {
 		// Empty quotes can count as a match, ignore them.
 		if match[1] == "" && match[2] == "" {
 			continue
 		}
-		// Get relevant match (one of these has to be a non-empty string, otherwise this wouldn't be
-		// a match).
+		// Get relevant match (one of these has to be a non-empty string, otherwise this wouldn't be a match).
 		matchString := match[1]
 		if matchString == "" {
 			matchString = match[2]
 		}
+		matchStrings[i] = matchString
```

Comment:
can we do:
```
if matchString != "" {
  matchStrings[i] = matchString
}
```
?

Reply:
actually I guess we don't need it because of the check above, we have a guarantee that both match[1] and match[2] are not empty strings.